docs/configuration.rst

```diff
@@ -24,21 +24,63 @@ Top level keys
 - ``title_format`` -- A format string for the title of your project. ``{name} {version} ({project_date})`` by default.
 - ``issue_format`` -- A format string for rendering the issue/ticket number in newsfiles. ``#{issue}`` by default.
 - ``underlines`` -- The characters used for underlining headers. ``["=", "-", "~"]`` by default.
+- ``fragment_types`` -- A list with the user custom fragment types. If empty or not provided, towncrier's default fragment types will be used.
 
 
 Custom fragment types
 ---------------------
+``Towncrier`` allows defining custom fragment types. Custom fragment types
+will be used instead ``towncrier`` default ones, they are not combined.
 
-Custom fragment types can be including in the
-pyproject.toml.
-Each custom type (``[[tool.towncrier.type]]``) has the following keys:
+Users can configure each of their own custom fragment types by adding tables to
+the pyproject.toml named ``[tool.towncrier.type.<a custom fragment type>]``.
+
+These tables may include the following optional keys:
+
+ * ``name``: The description of the fragment type, as it must be included
+   in the news file. If omitted, it defaults to  its  fragment type,
+   but capitalized.
+ * ``showcontent``: Whether if the fragment contents should be included in the
+   news file. If omitted, it defaults to ``true``
+
+
+For example:
+
+.. code-block:: toml
+
+
+    [tool.towncrier]
+    fragment_types = [
```

Comment:
Do you refer to moving all the fields in `[tool.towncrier]` to `[tool.towncrier.settings]` like this?:

```toml
[tool.towncrier]
[tool.towncrier.settings]
package="foo"
newsfile="news.rst"
fragment_types = []
```
The Json equivalent:
```json
{
  "tool": {
    "towncrier": {
      "settings": {
        "package": "foo",
        "newsfile": "news.rst",
        "fragment_types": []
      }
    }
  }
}

```

I don't think it solves anything. It just adds a nesting level.
---------------------------------------------------------------------------------------------------------------------

The conflicts that I mention was about  using `[tool.towncrier.chores]` :
```toml
[tool.towncrier]
package="foo"
newsfile="news.rst"
fragment_types = [
     "chores",
     "newsfile", # Same as the  field of [tool.towncier]
]
[tool.towncrier.chores]
name= "foo"

```
```json
{
  "tool": {
    "towncrier": {
      "package": "foo",
      "newsfile": "news.rst",
      "fragment_types": [
        "chore",
        "package"
      ],
      "chore": {
        "name": "foo",
      }
    }
  }
}
```
`"package"` was not configured, but another field is named like that so `self.config.get("package", {})` would return a str where a dict was expected. 
 If we  use `[tool.towncrier.fragment.chores]` (I can use other thing instead of `fragment` as  long as it is not used, `fragment_options` for example) this problem can't happen:
```toml
[tool.towncrier]
package="foo"
newsfile="news.rst"
fragment_types = [
     "chores",
     "newsfile", # Same as the  field of [tool.towncier]
]
[tool.towncrier.fragment.chores]
name= "foo"

```
```json
{
  "tool": {
    "towncrier": {
      "package": "foo",
      "newsfile": "news.rst",
      "fragment_types": [
        "chore",
        "package"
      ],
      "fragment": {
        "chore": {
          "name": "foo",
        }
      }
    }
  }
}
```
Since towncrier uses `"fragment"` to store all the options now, `self.config.get("fragment", {}).get("package", {})` (sorry for the trainwreck) now will return an empty dict.

Reply:
> I don't think it solves anything. It just adds a nesting level.

I was thinking that this allows having `towncrier.chore` fragment or  `towncrier.package`

The only "reserved" fragmet would be "settings".

-------

But we can play it safe and go with 

```
[tool.towncrier.fragment.chores]
name= "foo"
```